docs/design/storagenode-graceful-exit/pieces.md

```diff
@@ -0,0 +1,45 @@
+# Storage Node Graceful Exit - Gathering Pieces
+
+[Graceful Exit Overview](overview.md)
+
+## Abstract
+
+During Graceful Exit, a satellite needs to find pieces to be transferred from the exiting storage node.
+
+## Background
+
+Graceful Exit contains a process that moves existing pieces from one storage node to another. To accomplish this we need a list of pieces that need to be transferred.
+
+Pieces with lower durability have higher importance to be transferred.
+
+## Design
+
+We need a service on the satellite that finds pieces in the metainfo database that need to be transferred. We'll call this service `gexit.Service` or Graceful Exit service.
+
+The service starts by asking overlay for all exiting nodes where `nodes.exit_loop_completed` is false.
+
+Then joins a metainfo loop to iterate over all segments. For any segment that contains nodes that are exiting it will add an entry to a queue (if durability <= optimal). We call this the transfer queue. If durability > optimal, we remove the exiting node from the segment / pointer.
+
+
+The transfer queue is stored in database. We will need batching when inserting to database to avoid excessive load.
+
+Once metainfo loop has completed successfully it updates `nodes.exit_loop_completed` to indicate the storage nodes is ready for transferring.
+
+In the event that the satellite does not complete the metainfo loop (e.g. satellite is shutodwn), the service will re-enter the metainfo loop for all exiting nodes where `nodes.exit_loop_completed` is false. Pieces that already exist in the queue should not get duplicated.
```

Comment:
Here too: `nodes.exit_loop_completed` should be checked for `NULL`, not `false`.

Reply:
Updated